compiler/src/dotty/tools/repl/JLineTerminal.scala

```diff
@@ -92,11 +92,13 @@ final class JLineTerminal extends java.io.Closeable {
 
       context match {
         case ParseContext.ACCEPT_LINE =>
-          val lastLineOffset = line.lastIndexOfSlice(System.lineSeparator)
-          if (cursor <= lastLineOffset || ParseResult.isIncomplete(line)) incomplete()
+          // ENTER means SUBMIT when
+          //   - cursor is at end (discarding whitespaces)
+          //   - and, input line is complete
+          val cursorIsAtEnd = line.indexWhere(!_.isWhitespace, from = cursor) >= 0
```

Comment:
Should be called cursorNotAtEnd no? In any case, it's a bit confusing that the comment above describes the conditions for submitting, but the code checks the conditions for _not_ submitting, either the code logic or the comment should be changed.